src/core/tsi/transport_security_grpc.h

```diff
@@ -69,6 +80,10 @@ struct tsi_zero_copy_grpc_protector_vtable {
   tsi_result (*unprotect)(tsi_zero_copy_grpc_protector* self,
                           grpc_slice_buffer* protected_slices,
                           grpc_slice_buffer* unprotected_slices);
+  tsi_result (*unprotect_get_frame_size)(tsi_zero_copy_grpc_protector* self,
```

Comment:
I'm not loving the name, but don't have a concrete suggestion... maybe `unprotect_and_get_min_progress_size` -- I do explicitly think it should be min_progress_size and not frame_size -- frame_size implies this is only for the body of a frame, but my expectation is that a protocol with a four byte header would set min_progress_size to four prior to seeing the next header.

Reply:
Renamed it now.